foundry.md

```diff
@@ -498,28 +451,22 @@ This rule then loads the storage slot identified by `#asWord(#range(LM, ARGSTART
 function store(address account, bytes32 slot, bytes32 value) external;
 ```
 
-`call.store` will match when the `store` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
-This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32))` and the new slot value using `#asWord(#range(LM, ARGSTART +Int 68, 32))` and updates the slot denoted by `#asWord(#range(LM, ARGSTART +Int 36, 32))`.
+`foundry.call.store` will match when the `store` cheat code function is called.
+This rule then takes from the function call data the account using `#asWord(#range(ARGS, 0, 32))` and the new slot value using `#asWord(#range(ARGS, 32, 32))` and updates the slot denoted by `#asWord(#range(ARGS, 64, 32))`.
 
-```k
-    rule [call.store]:
-         <k> CALL _ CHEAT_ADDR _VALUE ARGSTART _ARGWIDTH _RETSTART _RETWIDTH => #loadAccount #asWord(#range(LM, ARGSTART +Int 4, 32)) ~> #setStorage #asWord(#range(LM, ARGSTART +Int 4, 32)) #asWord(#range(LM, ARGSTART +Int 36, 32)) #asWord(#range(LM, ARGSTART +Int 68, 32)) ~> #finishCheatCode ... </k>
-         <localMem> LM </localMem>
+```{.k .bytes}
+    rule [foundry.call.store]:
+         <k> #call_foundry SELECTOR ARGS => #loadAccount #asWord(#range(ARGS, 0, 32)) ~> #setStorage #asWord(#range(ARGS, 0, 32)) #asWord(#range(ARGS, 32, 32)) #asWord(#range(ARGS, 64, 32)) ... </k>
          <output> _ => .ByteArray </output>
-      requires CHEAT_ADDR ==Int #address(FoundryCheat)
-       andBool #asWord(#range(LM, ARGSTART, 4)) ==Int 1892290747 // selector ( "store(address,bytes32,bytes32)" )
-      [priority(40)]
-
-    syntax KItem ::= "#setStorage" Int Int Int [klabel(foundry_setStorage)]
- // -----------------------------------------------------------------------
-    rule <k> #setStorage ACCTID LOC VALUE => . ... </k>
-         <account>
-           <acctID> ACCTID </acctID>
-           <storage> STORAGE => STORAGE [ LOC <- VALUE ] </storage>
-             ...
-         </account>
+      requires SELECTOR ==Int selector ( "store(address,bytes32,bytes32)" )
 ```
 
+Otherwise, ignore any other call to the Foundry contract.
+
+```{.k .bytes}
+    rule [foundry.call.owise]:
+         <k> #call_foundry _ _ => . ... </k> <output> _ => .ByteArray </output> [owise]
+```
```

Comment:
We shouldn't ignore, we should just halt. So we can either just remove this rule, or have the `owise` rule go to a special `#error_foundry String` production, which holds the function selector that could not match and says "not implemented".


If we add such an error state, we can go through and add explicit error rules for all the unimplemented cheatcodes:

```
rule <k> #call_foundry NAME _ => #unimplemented_foundry STRING ... </k>
    requires NAME ==Int selector(STRING)
```

That way we get better error messages for the methods that are proper cheat codes, and we just have not implemented yet, and we have a record directly in this file of all the unimplemented cheat codes.

Then we can have one rule for unknown cheatcodes:

```
rule <k> #call_foundry NAME _ => #unimplemented_foundry ("UNKNOWN: " +String Int2String(NAME)) ... </k>
```

Reply:
For now I just defined the `#error_foundry` production, will open a new issue for the handling of unimplemented cheat codes.